CIP-1694/README.md

```diff
@@ -0,0 +1,640 @@
+
+**At most one** governance action **of each type** (i.e. one from each of the six categories listed above) can be staged for enactment in any given epoch. For example, there can only be one treasury withdrawal action in a single epoch (it may, however, comprise many individual withdrawals).
+
+Once ratified, actions will be staged for enactment. Actions that have been staged will be enacted on the following epoch boundary, unless they are dropped. All submitted governance actions will therefore either:
+
+1. be **ratified**;
+2. be **dropped** as a result of some higher priority action; or else
+3. will **expire** after a number of epochs.
+
+Deposits are returned immediately when either:
+
+1. a ratified action is **enacted**;
+1. the action **expires**; or
+1. a ratified action is **dropped**.
+
+Ratification is explained in more detail later in this document.
+
+>**Note:**
+>This design means that the first governance action of a given type to be ratified will be staged for enactment in an epoch.
+
+#### Enactment
+
+Ratified actions may be enacted at an epoch boundary. During enactment, actions in the staging group for the current epoch are prioritized as follows:
+
+1. Motion of no-confidence;
+2. New Constitutional Committee or a change to the quorum size;
+3. Updates to the Constitution;
+4. Hard Fork initiation;
+5. Protocol parameter changes;
+6. Treasury withdrawals.
+
+As described above, at most one action of each type may be enacted in any given epoch.
+So a ratified _"Motion of no-confidence"_ action will be enacted first if there is one, and so on.
+
+A successful _"Motion of no-confidence"_, the election of a new Constitutional Committee, or a constitutional change invalidates *all* other **unenacted** governance actions (whether or not they have been ratified),
+_causing them to be immediately **dropped** without ever being enacted_. Deposits for dropped actions will be returned immediately.
+
+#### Content of a Governance Action
+
+Every governance action will include the following:
+
+* a deposit amount (recorded since the amount is an updatable protocol parameter);
+* a reward address to receive the repaid deposit;
+* a URL to any metadata that is needed to justify the action;
+* a hash of the contents of this metadata URL.
+
+// TODO: Provide a CBOR specification in the annexe for this new on-chain entity
+
+Additionally, the action will include the following:
+
+1. For protocol parameter changes - the changed parameters;
+2. For hard fork initiation - the new major protocol version, which must be one greater than the current version;
+3. For treasury withdrawals - a map from stake credentials to a positive number of Lovelace;
+4. For updates to the Constitution - a 32-byte blake2b-256 hash digest of the Constitution document;
+5. For new Constitutional Committee and changes to the quorum size - a set of key hashes and a positive number that is no greater than the size of the committee;
```

Comment:
@JaredCorduan we've talked about the size of the committee being a fraction instead, and I think I remember convincing you that this is a good idea. Any reason why it's not in here?

As a reminder, and for anyone else, the reason why I believe this is better is purely psychological. We generally think of some percentage majority required for something to pass, and changing that percentage is very different from changing the people involved. But if this is just a number, one could imagine a committee that grows over time but doesn't adjust the quorum number properly, thus weakening the required majority. With a fraction, this would be much more obvious to everyone.

Reply:
apologies, that was not intentional. I believe @dcoutts wanted to advocate for using a natural number. Ideally, the quorum should never be changed without considering a change to the quorum size, in either case, in which case the natural might be more explicit and clear.

Reply:
It's not that I wouldn't expect people to not consider the quorum size while changing the committee size, it's that the default is wrong. The default is always not to change this value, which works fine if it's a fraction, but doesn't otherwise. And you generally have to argue harder to convince people not to do the default thing. So with a number I'd expect the quorum to drop to ½ eventually, no matter what the initial value is, whereas with a fraction it's much harder do do this sort of thing sneakily.

Reply:
Also, how would people go about determining an appropriate size of the quorum? Most likely by targeting some fraction, say ⅔ and picking the quorum number to match that fraction. This just gives an additional layer for people to play politics with.

Actually, let me give an example. Let's say we target requiring a ⅔ majority. We currently have 9 people on the committee, with a required quorum of 6. Now, we add two people and increase the quorum by 1, which makes sense. We let some time pass, and at some point we want to add another two people to the committee. What should the quorum be? Well, last time we did that, we just increased the quorum by 1, so we can play the game and argue that we should only increase it by one again, and people might not be paying enough attention to notice.

However, if it was a fraction instead, it would have stayed at ⅔ all the time, and the most recent proposal would have to drop the fraction to around 60% instead to have the same effect on the quorum. That's super obvious to everyone, and clearly unprecedented.

Reply:
When the committee changes, we would force the quorum to be specified as well, so there would be no default, only the memory of the previous one. But I guess you are saying that there is a pathological gravity to explicitly keeping the same quorum even when you see the group get bigger?